symengine/polynomial.h

```diff
@@ -96,11 +170,159 @@ RCP<const UnivariatePolynomial> sub_uni_poly(const UnivariatePolynomial &a, cons
+    }
+    
+    UnivariateExprPolynomial &operator+=(const UnivariateExprPolynomial &other)
+    {
+        poly_ = add_uni_poly(*(poly_.ptr()), *(other.poly_.ptr()));
+        return *this;
+    }
+    
+    friend UnivariateExprPolynomial operator-(const UnivariateExprPolynomial &a, const UnivariateExprPolynomial &b)
+    {
+        return UnivariateExprPolynomial(sub_uni_poly(*(a.poly_.ptr()), *(b.poly_.ptr())));
+    }
+    
+    UnivariateExprPolynomial operator-() const
+    {
+        UnivariateExprPolynomial retval(*this);
+        neg_uni_poly(*(retval.poly_.ptr()));
+        return retval;
+    }
+    
+    UnivariateExprPolynomial &operator-=(const UnivariateExprPolynomial &other)
+    {
+        poly_ = sub_uni_poly(*(poly_.ptr()), *(other.poly_.ptr()));
+        return *this;
+    }
+    
+    friend UnivariateExprPolynomial operator*(const UnivariateExprPolynomial &a, const UnivariateExprPolynomial &b)
+    {   
+        return UnivariateExprPolynomial(mul_uni_poly(a.poly_, b.poly_));
+    }   
+    
+    friend UnivariateExprPolynomial operator/(const UnivariateExprPolynomial &a, const Expression &b)
+    {
+        return UnivariateExprPolynomial(mul_uni_poly(a.poly_, UnivariateExprPolynomial(1/b).poly_));    
+    }
+        
+    UnivariateExprPolynomial &operator*=(const UnivariateExprPolynomial &other)
+    {   
+        poly_ = mul_uni_poly(poly_, other.poly_);
+        return *this;
+    }   
+    
+    bool operator==(const UnivariateExprPolynomial &other) const
+    {   
+        return eq(*poly_, *other.poly_);
+    }   
+    
+    bool operator==(int i) const
+    {
+        return eq(*poly_, *(UnivariateExprPolynomial(i).poly_)); 
+    }
+     
+    bool operator!=(const UnivariateExprPolynomial &other) const
+    {   
+        return not (*this == other);
+    }   
+    
+    //! Method to get UnivariatePolynomial from UnivariateExprPolynomial
+    const RCP<const UnivariatePolynomial>& get_univariate_poly() const
+    {   
+        return poly_;
+    }   
+    
+    std::size_t __hash__() const
+    {
+        return (poly_.ptr())->__hash__();
+    }
+    
+    const RCP<const Basic> get_basic() const
+    {
+        RCP<const Basic> basic;
+        return std::move(basic);
+        //return make_rcp<const Symbol>("x");//make_rcp<const UnivariatePolynomial>(poly_->var_, poly_->get_args());
+    }
+
+    int compare(const UnivariateExprPolynomial &other)
+    {
+       return eq(*poly_, *other.poly_); 
+    }
+}; //UnivariateExprPolynomial
+
+
+class sym_hash {
+public:
+  size_t operator()(const Symbol &s) const {
+    return s.__hash__();
+  }
+};
+
+class sym_compare {
+  public:
+  size_t operator()(const Symbol &a, const Symbol &b){
+    return a.compare(b);
+  }
+};
+
+class sym_eq {
+ public:
+  bool operator()(const Symbol &a, const Symbol &b){
+    return a.__eq__(b);
+  }
+};
+
+int umap_vec_mpz_compare(umap_vec_mpz &a, umap_vec_mpz &b){
```

Comment:
change this to `inline int umap_vec_mpz_compare(umap_vec_mpz &a, umap_vec_mpz &b){`